fl4health/datasets/skin_cancer/load_data.py

```diff
@@ -0,0 +1,204 @@
+
+    dataset_paths = {
+        "Barcelona": data_dir / "ISIC_2019" / "ISIC_19_Barcelona.json",
+        "Rosendahl": data_dir / "HAM10000" / "HAM_rosendahl.json",
+        "Vienna": data_dir / "HAM10000" / "HAM_vienna.json",
+        "UFES": data_dir / "PAD-UFES-20" / "PAD_UFES_20.json",
+        "Canada": data_dir / "Derm7pt" / "Derm7pt.json",
+    }
+
+    if dataset_name not in dataset_paths:
+        log(ERROR, f"Dataset {dataset_name} not found in available datasets.")
+        raise ValueError(f"Dataset {dataset_name} not found in available datasets.")
+
+    dataset_path = dataset_paths[dataset_name]
+
+    if not dataset_path.exists():
+        log(
+            ERROR,
+            f"Dataset file {dataset_path} does not exist.\
+            Please follow the instructions in fl4health/datasets/skin_cancer/README.md.",
+        )
+        raise FileNotFoundError(f"Dataset file {dataset_path} does not exist.")
+
+    log(INFO, f"Data directory: {str(dataset_path)}")
+
+    with open(dataset_path, "r") as f:
+        data = json.load(f)["data"]
+
+    random.seed(seed)
+    random.shuffle(data)
+    total_size = len(data)
+    train_size = int(0.7 * total_size)
+    valid_size = int(0.15 * total_size)
+
+    train_data = data[:train_size]
+    valid_data = data[train_size : train_size + valid_size]
+
+    if train_transform is None:
+        train_transform = transforms.Compose(
+            [
+                transforms.RandomHorizontalFlip(),
+                transforms.RandomVerticalFlip(),
+                transforms.RandomRotation(20),
+                transforms.ColorJitter(brightness=32.0 / 255.0, saturation=0.5),
+                transforms.Resize([256, 256]),
+                transforms.ToTensor(),
+                transforms.Normalize((0.0, 0.0, 0.0), (1.0, 1.0, 1.0)),
+            ]
+        )
+
+    if val_transform is None:
+        val_transform = transforms.Compose(
+            [
+                transforms.Resize([256, 256]),
+                transforms.ToTensor(),
+                transforms.Normalize((0.0, 0.0, 0.0), (1.0, 1.0, 1.0)),
+            ]
+        )
+
+    train_ds: TensorDataset = construct_skin_cancer_tensor_dataset(train_data, transform=train_transform)
+    valid_ds: TensorDataset = construct_skin_cancer_tensor_dataset(valid_data, transform=val_transform)
+
+    if sampler is not None:
+        train_ds = sampler.subsample(train_ds)
+        valid_ds = sampler.subsample(valid_ds)
+
+    if dataset_converter is not None:
+        train_ds = dataset_converter.convert_dataset(train_ds)
+        valid_ds = dataset_converter.convert_dataset(valid_ds)
+
+    train_loader = DataLoader(train_ds, batch_size=batch_size, shuffle=True)
+    validation_loader = DataLoader(valid_ds, batch_size=batch_size)
+
+    num_examples = {"train_set": len(train_ds), "validation_set": len(valid_ds)}
+
+    return train_loader, validation_loader, num_examples
+
+
+def load_skin_cancer_test_data(
```

Comment:
Rather than having a separate function to load the test data, I think we can just fold this into the load function above. That way a user doesn't have to be "careful" about how the shuffles happen between the two separate functions to ensure mutual exclusivity. One of the arguments can just be the split percentages as a tuple (0.7, 0.15, 0.15) or the like?

Reply:
I added an argument as well as a check to see if the values provided sum to 1